### ISSUE TITLE
clusterCommander refactor

### ISSUE THREAD

**Issue by tuan08** [DemandCube/NeverwinterDP_Legacy#681]
The script can be improved in many way:
1.  Respect the object oriented design and code. the getReport method does not belong to any class.
2.  In the ServerSet class ,  you can have a method that get sub set of the server set like:
   getHadoopServerSet(), getZookeeperServerSet() ....
3.  Have a specific class for each set like HadoopServerSet, ZookeeperServerSet...
4. Have find method and many of your code can do like  serverSet.findServer('name').findProcess('name').kill() instead of repeat the loop
5. In the serverset class ,  you can have the method  like  hadoopServerSet.getRandom().kill() 

There is many more way to improve , try to figure out yourself

**Comment by tuan08** [comment]
I see many of your code is in the ServerSet class and it is not a good organization.  You should write the good code a part from making it works

**Comment by tuan08** [comment]
Most of your code need to iterate through each server , each process to check. Make sure you have:
1.  HadoopServerSet, ZookeeperServerSet.... implementation 
2. in the main ServerSet , you can have a method  getServerSetByRole('role') that return Hadoop, Zookeeper, Kafka ... set
3. Try to have have the generic method like  find() , getRandom() ...  and you can have serverSet.find().kill() .. , serverSet.getRandom().kill() ...

**Comment by tuan08** [comment]
startZookeeper(),  shutdownZookeeper() .... should be  findProcess('zookeeper').start() , findProcess('zookeeper').shutdown()

**Comment by tuan08** [comment]
Try to have zookeeper, kafka ... as enum or  constant in java

**Comment by KingOfPoptart** [comment]
Lets work on this ticket when we work on moving clusterCommander to a private repo.  I'm sure there will be fixes needed when we move the code over, so lets do these improvements at the same time

**Comment by jeroldleslie** [comment]
Yes. I agree with @KingOfPoptart . And the comments by @tuan08 make the code much better. This should be done when we work on moving clusterCommander to a private repo.

**Comment by KingOfPoptart** [comment]
What @tuan08 means about enums or constants - 

Instead of having magic strings like "zookeeper", "kafka", "QuorumPeerMain", etc - Move them all to constants.  In Python you'll have to do something like

```
class Constants(): 
  zookeeperProcessIdentifier = "QuorumPeerMain"
  otherConstant = "xxxx"
```

And then in some other file:

```
import Constants
class OtherClass(): 
  def x(): 
    string = Constants.zookeeperProcessIdentifier
```
